lib/content-services/src/lib/search/components/search-properties/search-properties.component.ts

```diff
@@ -0,0 +1,216 @@
+
+@Component({
+    selector: 'adf-search-properties',
+    templateUrl: './search-properties.component.html',
+    styleUrls: ['./search-properties.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class SearchPropertiesComponent implements OnInit, AfterViewChecked, SearchWidget {
+    id: string;
+    settings?: SearchWidgetSettings;
+    context?: SearchQueryBuilderService;
+    startValue: SearchProperties;
+    displayValue$ = new Subject<string>();
+
+    private _form = this.formBuilder.nonNullable.group<FileSizeCondition>({
+        fileSizeOperator: FileSizeOperator.AT_LEAST,
+        fileSize: undefined,
+        fileSizeUnit: FileSizeUnit.KB
+    });
+    private _fileSizeOperators = Object.keys(FileSizeOperator).map<string>(key => FileSizeOperator[key]);
+    private _fileSizeUnits = [FileSizeUnit.KB, FileSizeUnit.MB, FileSizeUnit.GB];
+    private canvas = document.createElement('canvas');
+    private _fileSizeOperatorsMaxWidth: number;
+    private _selectedExtensions: string[];
+    private _reset$ = new Subject<void>();
+    private sizeField: string;
+    private nameField: string;
+
+    @ViewChild('fileSizeOperatorSelect', {read: ElementRef})
+    fileSizeOperatorSelectElement: ElementRef;
+
+    get form(): SearchPropertiesComponent['_form'] {
+        return this._form;
+    }
+
+    get fileSizeOperators(): string[] {
+        return this._fileSizeOperators;
+    }
+
+    get fileSizeUnits(): FileSizeUnit[] {
+        return this._fileSizeUnits;
+    }
+
+    get fileSizeOperatorsMaxWidth(): number {
+        return this._fileSizeOperatorsMaxWidth;
+    }
+
+    get reset$(): Subject<void> {
+        return this._reset$;
+    }
+
+    set selectedExtensions(extensions: string[]) {
+        this._selectedExtensions = extensions;
+    }
+
+    constructor(private formBuilder: FormBuilder, private translateService: TranslateService) {}
+
+    ngOnInit() {
+        if (this.settings) {
+            if (!this.settings.fileExtensions) {
+                this.settings.fileExtensions = [];
+            }
+            [this.sizeField, this.nameField] = this.settings.field.split(',');
+        }
+        if (this.startValue) {
+            this.setValue(this.startValue);
+        }
+    }
+
+    ngAfterViewChecked() {
+        if (this.fileSizeOperatorSelectElement?.nativeElement.clientWidth && !this._fileSizeOperatorsMaxWidth) {
+            setTimeout(() => {
+                const extraFreeSpace = 20;
+                this._fileSizeOperatorsMaxWidth = Math.max(...this._fileSizeOperators.map((operator) =>
+                    this.getOperatorNameWidth(operator, this.getCanvasFont(this.fileSizeOperatorSelectElement.nativeElement)))) +
+                    this.fileSizeOperatorSelectElement.nativeElement.querySelector('.mat-select-arrow-wrapper').clientWidth +
+                    extraFreeSpace;
+            });
+        }
+    }
+
+    narrowDownAllowedCharacters(event: Event) {
```

Comment:
it is the same approach like for created/modified dates:

https://github.com/Alfresco/alfresco-ng2-components/pull/8714/files#diff-3d9b6561e3d273c811ef01fd21125aa8beafdbf53e18815843346c45a40d27b1R167

Basically pattern would mark this as invalid but we don't want to even allow to type incorrect character because we don't want to display error message at all. We just want to prevent typing incorrect characters. Pattern will allow type incorrect characters and just mark field as invalid